File: Bank/app/src/main/java/com/msbm/bank/loginScreen/LoginConfigurator.java
package com.msbm.bank.loginScreen;

import java.lang.ref.WeakReference;

public enum LoginConfigurator {
    INSTANCE;
    public void configure(LoginActivity loginActivity) {
        LoginPresenter loginPresenter = new LoginPresenter();
        loginPresenter.loginActivity = new WeakReference<>(loginActivity);

        LoginInteractor loginInteractor = new LoginInteractor();
        loginInteractor.loginPresenterInput = loginPresenter;

        if (loginActivity.loginInteractor == null) {
            loginActivity.loginInteractor = loginInteractor;
        }
    }
}

File: app/src/main/java/io/github/maikotrindade/bankapp/login/domain/LoginConfigurator.kt
package io.github.maikotrindade.bankapp.login.domain

import io.github.maikotrindade.bankapp.base.ui.MainActivity
import io.github.maikotrindade.bankapp.login.ui.LoginFragment
import java.lang.ref.WeakReference

enum class LoginConfigurator {

    INSTANCE;

    fun configure(fragment: LoginFragment) {

        //config router
        val router = LoginRouter()
        router.activity = WeakReference(fragment.activity as MainActivity)
        fragment.router = fragment.router ?: router

        //config interactor and callbacks
        val interactor = LoginInteractor()
        interactor.presenter = LoginPresenter()
        interactor.presenter.fragmentInput = WeakReference(fragment)
        fragment.interactor = fragment.interactor ?: interactor

    }

}